lambda/rotator-snowflake/setup/setup.go

```diff
@@ -2,42 +2,130 @@ package setup
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"github.com/chanzuckerberg/go-misc/databricks"
 	"github.com/chanzuckerberg/go-misc/errors"
 	databricksCfg "github.com/chanzuckerberg/go-misc/lambda/rotator-snowflake/setup/databricks"
 	oktaCfg "github.com/chanzuckerberg/go-misc/lambda/rotator-snowflake/setup/okta"
 	snowflakeCfg "github.com/chanzuckerberg/go-misc/lambda/rotator-snowflake/setup/snowflake"
 	"github.com/chanzuckerberg/go-misc/sets"
+	"github.com/chanzuckerberg/go-misc/snowflake"
+	"github.com/hashicorp/go-multierror"
 	"github.com/kelseyhightower/envconfig"
+	"github.com/okta/okta-sdk-golang/okta"
+	"github.com/segmentio/chamber/store"
 )
 
-func Okta(ctx context.Context) (*oktaCfg.OktaClient, error) {
-	return oktaCfg.GetOktaClient(ctx)
+type SecretStore interface {
+	Read(id store.SecretId, version int) (store.Secret, error)
 }
 
-func Databricks(ctx context.Context) (*databricksCfg.Account, error) {
+func Okta(ctx context.Context, secrets SecretStore) (*oktaCfg.OktaClient, error) {
+	env, err := oktaCfg.LoadOktaClientEnv()
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to load right Okta env variables")
+	}
+
+	service := env.PARAM_STORE_SERVICE
+	tokenSecretID := store.SecretId{
+		Service: service,
+		Key:     "okta_private_key",
+	}
+	private_key, err := secrets.Read(tokenSecretID, -1)
```

Comment:
know what this -1 is? worth making it a var to contextualize it?

Reply:
Oh, I think I'll need to look over the code & find other instances of magic #s. This is an index that gets the latest version of a secret. I'll make this a var!

Reply:
let's fix that instead of this! taking it offline